### DISCUSSION TITLE
Some issues regarding load and generator output replacement in the case.

### DISCUSSION THREAD

**Discussion by Crazy-Bike** [CURENT/andes#536] [Q&A]
Hello!
I am trying to put the results of unit combination optimization scheduling into Andes for simulation to compare and verify my transient simulation results. I replaced the data on the IEEE-39 case by replacing the p0 and q0 of the PQ model to simulate different loads on nodes; Replacing the p0 and q0 of the PV model to simulate different outputs of each generator node did not result in successful Andes simulation. (The cases before and after the change all used the REGCV1 model to replace some GENROU units)
Figure 1 shows the frequency variation curve of the system after disturbance under the raw data of IEEE-39 case, which looks very accurate; But Figure 2 shows the frequency change curve of the system after being disturbed, which looks very strange.
May I ask what I need to pay attention to when simulating different load scenarios (different load levels will correspond to different power flows and generator outputs, that is, p0 and q0 in PQ and PV models in Excel are changing)? 
Is it feasible to keep the reactive power unchanged and only change the value of active power if only studying frequency?
Thanks!
![image](https://github.com/CURENT/andes/assets/130566375/73444cd5-66bd-4f80-a217-4f45c4e6b8a0)
![image](https://github.com/CURENT/andes/assets/130566375/0f5135d7-d7f5-4a92-b063-18a4faf53a91)
[ieee39_VSG01.xlsx](https://github.com/CURENT/andes/files/15097177/ieee39_VSG01.xlsx)

**Comment by Crazy-Bike** [comment]
Like these:
![image](https://github.com/CURENT/andes/assets/130566375/ff00c264-5d51-4275-b2a3-ab3c14a6d0dd)
![image](https://github.com/CURENT/andes/assets/130566375/649b48c2-9a2a-47e1-be6e-86ac552b76f6)

**Comment by jinningwang** [comment]
> May I ask what I need to pay attention to when simulating different load scenarios (different load levels will correspond to different power flows and generator outputs, that is, p0 and q0 in PQ and PV models in Excel are changing)?

Depends on the cases. In ANDES, PV bus voltage are determined by the connected PV generator. Also, for PV generator, setting q0 will not work given the power flow algorithm.

Check the PQ load settings as it might impact the frequency response.

> Is it feasible to keep the reactive power unchanged and only change the value of active power if only studying frequency?

For simulation, yes. For frequency study, depends on what level of details you need.

There is an interface between AMS and ANDES, which is designed to broadcast the optimization results. Check the code for more details if necessary.
https://github.com/CURENT/ams/blob/fcc7e579c8f7453fae3ddebbd0eedbe4e463c1c9/ams/interop/andes.py#L517-L638
